libglusterfs/src/mem-pool.c

```diff
@@ -463,29 +456,38 @@ pool_sweeper(void *arg)
     pending = true;
 
     for (;;) {
+        int ret;
+        struct timespec ts;
+
+        timespec_now_realtime(&ts);
         /* If we know there's pending work to do (or it's the first run), we
          * do collect garbage more often. */
-        sleep(pending ? POOL_SWEEP_SECS / 5 : POOL_SWEEP_SECS);
+        ts.tv_sec += (pending ? POOL_SWEEP_SECS / 5 : POOL_SWEEP_SECS);
+
+        pthread_mutex_lock(&sweeper_lock);
+        ret = pthread_cond_timedwait(&sweeper_cond, &sweeper_lock, &ts);
```

Comment:
We can ignore the return value of `pthread_cond_timedwait()` and read the value of `sweeper_quit` instead.

Reply:
OK